src/components/Login.js

```diff
@@ -40,123 +40,117 @@ const themeColors = {
   4: ['#FF94DB', 'white'],
 };
 
-class Login extends React.Component {
-  state = {
-    dummy: false,
-    index: Math.floor(Math.random() * 5),
-  };
+function Login (props) {
+ const [dummy, setDummy] = useState(false);
+ const [index, setIndex] = useState(Math.floor(Math.random() * 5));
 
-  // basically, when the window resizes, we should recalculate get SVG - the window parameters change!
-  componentDidMount = () => {
-    window.addEventListener('resize', () => this.setState({ dummy: !this.state.dummy }));
-  };
+ useEffect(() => {
+   window.addEventListener('resize', () => setDummy(!dummy));
+   return () => {
+     window.removeEventListener('resize', () => setDummy(!dummy));
+   };
+ });
```

Comment:
This should be useEffect(func, [])

Reply:
If you wanted to know why the array is needed:

- Giving it an empty array acts like componentDidMount as in, it only runs once. (**this is what we want**)
- Giving it no second argument acts as both componentDidMount and componentDidUpdate, as in it runs first on mount and then on every re-render.
- Giving it an array as second argument with any value inside, eg , [variable1] will only execute the code inside your useEffect hook ONCE on mount, as well as whenever that particular variable (variable1) changes.